### PR TITLE
Prefer native `href` links where possible

### PR DESCRIPTION
**Pull request by Pinjasaur**
#### Summary
I manually audited instances of `navigateToUrl` _and_ `navigateToPluginUrl` to see where it would be straightforward to drop-in a native `<a href="...">`-style link. The benefit being native hyperlinks allow for "open in new tab" workflows which is a better UX.

Unfortunately, many of the programmatic `onClick`-style still remain. This is due to the notion of "changing the URL" to be tied to other things e.g. `dispatch`-ing the Redux store and/or firing off telemetry events. Also, many of the components are styled at a high-level to be `<button>`s and swapping in a `<a>` isn't a 1:1.

I think it's certainly do-able to swap over the majority of `navigate{Plugin}ToUrl` instances to be `href`s but would be a considerably larger task given the need to refactor some of the components to _accept_ an `href` prop.

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/19169

#### Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
~~- [ ] Unit tests updated~~

### CODE REVIEW

**Review by agarciamontoro — COMMENTED** [review]
Thank you for taking a look at this, @Pinjasaur, it's a lot of work to review those one by one. The code you added looks good, although the `RHSListView` component seems to be unused code, actually. I'll open a PR to remove it.

Did you have a chance to look at the Run details link in the RHS header (the one in the screenshot)? If so, what was the problem with that one?

image

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks, @Pinjasaur, especially for the overview on the other places in the code where these changes weren't as straightforward. A few comments below.

**Comment by agarciamontoro** [comment]
@Pinjasaur, the PR #1018 created some conflicts here, as expected. If you're ok with it, I can fix them myself tomorrow :)

**Comment by Pinjasaur** [comment]
> @Pinjasaur, the PR #1018 created some conflicts here, as expected. If you're ok with it, I can fix them myself tomorrow :)

Good with me—thanks @agarciamontoro. Aiming to get the rest of the feedback on this PR buttoned up by then. 🙂

**Comment by Pinjasaur** [comment]
> The code you added looks good, although the `RHSListView` component seems to be unused code, actually. I'll open a PR to remove it.

Good catch, thanks.

> Did you have a chance to look at the Run details link in the RHS header (the one in the screenshot)? If so, what was the problem with that one?
> 
> image

I revisited this with fresh eyes and was able to come up with a reasonable solution by creating a `<RHSTitleLink>` component based off of the `<RHSTitleText>` but styles a `<Link>` instead of a `<button>`. I also took the opportunity to remove a `fill: ...` CSS statement as those are only relevant for SVG, not 'normal' HTML elements.

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx [hunk 1]
```diff
@@ -392,24 +388,28 @@ const PlaybookRunBackstage = () => {
                         </TitleWithBadgeAndLink>
                         {
                             playbook &&
-                            <PlaybookDiv onClick={() => navigateToPluginUrl(`/playbooks/${playbook?.id}`)}>
+                            <Link
```
[the remark this answers is not in the document]
Comment (Pinjasaur) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx]: Sorry, realized I forgot to remove the now-unused `PlaybookDiv`. 🤦🏻

I used the same strategy as `RHSTitleLink` and renamed `PlaybookDiv` -> `PlaybookLink` which is a styled `Link`. This also removes those 2 inlined CSS statements.

File: server/app/playbook_run_service.go [hunk 2]
```diff
@@ -755,7 +755,7 @@ func (s *PlaybookRunServiceImpl) UpdateStatus(playbookRunID, userID string, opti
 		PlaybookRunID: playbookRunID,
 		PostID:        channelPost.Id,
 	}); err != nil {
-		return errors.Wrap(err, "failed to write status post to store. There is now inconsistent state.")
+		return errors.Wrap(err, "failed to write status post to store. There is now inconsistent state")
```
[the remark this answers is not in the document]
Comment (Pinjasaur) [reply server/app/playbook_run_service.go:758]: Yeah, 100% agree. I tried to be proactive by having the change as a single commit but I also realize, as you mentioned, it's not a related change. My reasoning was `make check-style` wasn't working without the change and that was documented as one of the pre-reqs for opening a PR.

What's the recommendation? Off the top of my head, I could cherry pick that commit into a separate PR and revert it from _this_ PR.

**Comment by agarciamontoro** [comment]
@Pinjasaur, heads-up that I merged master, removing the file that was deleted in #1018 and fixing a conflict with the error message in playbook_run_service, which was actually fixed in master a couple of days ago, in #937 :)

File: server/app/playbook_run_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply server/app/playbook_run_service.go]: This was actually fixed in #937, so no need to do anything else here!

File: webapp/src/components/rhs/rhs_title.tsx [hunk 3]
```diff
@@ -111,11 +104,36 @@ const RHSTitleText = styled.div<{ clickable?: boolean }>`
...
     `}
 `;
 
+const RHSTitleLink = styled(Link)`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    padding: 0 4px;
+    color: var(--center-channel-color) !important;
```
Comment (agarciamontoro) [inline webapp/src/components/rhs/rhs_title.tsx:117]: Do we need the `!important` here? If so, styled-components usually tackle these type of issues by incrementing specificity by doing something like
```css
&&& {
    color: var(--center-channel-color) !important;
}
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Looks great, thank you for all your work on this, @Pinjasaur!

I only left one comment below about the use of important in the CSS.

File: server/app/playbook_run_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lieut-data) [reply server/app/playbook_run_service.go]: Oh wow, I had no idea `make check-style` was broken here -- your reasoning for including totally made sense. Thanks, @Pinjasaur!

**Review by lieut-data — APPROVED** [review]

File: webapp/src/components/rhs/rhs_title.tsx (hunk 3 above)
Comment (Pinjasaur) [reply webapp/src/components/rhs/rhs_title.tsx:117]: Thanks for the suggestion. `&&` did the trick. 🙂

**Review by agarciamontoro — APPROVED** [review]
I was going to say that we needed three `&`, not two, but you beat me to it!

Great work, @Pinjasaur, thank you again for your contribution!

**Pull request merged**